Runtime/Scripts/BaseMeshSync.Modifiers.cs

```diff
@@ -0,0 +1,544 @@
+                return SaveArray("a", intArray);
+            else if (valueToSerialize is float[] floatArray)
+                return SaveArray("b", floatArray);
+            else
+                throw new NotImplementedException($"propertyValue: {valueToSerialize.GetType()} cannot be serialized!");
+        }
+
+        static object DeserializeString(string serializedValue)
+        {
+            if (serializedValue.Length == 0)
+                return null;
+            char type = serializedValue[0];
+            if (type == 'n')
+                return null;
+            else if (type == 'i')
+                return int.Parse(serializedValue.Substring(1));
+            else if (type == 'f')
+                return float.Parse(serializedValue.Substring(1));
+            else if (type == 's')
+                return serializedValue.Substring(1);
+            else if (type == 'a')
+                return ParseArray(serializedValue, int.Parse);
+            else if (type == 'b')
+                return ParseArray(serializedValue, float.Parse);
+            else
+                throw new NotImplementedException($"propertyValue: {type} cannot be deserialized!");
+        }
+
+        public void SetSerializedValue(string serializedValue)
+        {
+            NewValue = DeserializeString(serializedValue);
+        }
+
+        public void OnBeforeSerialize()
+        {
+            propertyValueSerialized = GetSerializedValue();
+        }
+
+        public void OnAfterDeserialize()
+        {
+            propertyValue = DeserializeString(propertyValueSerialized);
+        }
+
+        public object NewValue
+        {
+            get => newValue;
+            set
+            {
+                switch (type)
+                {
+                    case PropertyInfoDataType.IntArray:
+                        {
+                            var newValueAsArray = new int[arrayLength];
+
+                            if (value is Vector2 newValueAsVector2)
+                            {
+                                newValueAsArray[0] = Mathf.Clamp((int)newValueAsVector2.x, (int)min, (int)max);
+                                newValueAsArray[1] = Mathf.Clamp((int)newValueAsVector2.y, (int)min, (int)max);
+
+                                value = newValueAsArray;
+                            }
+                            else if (value is Vector3 newValueAsVector3)
+                            {
+                                newValueAsArray[0] = Mathf.Clamp((int)newValueAsVector3.x, (int)min, (int)max);
+                                newValueAsArray[1] = Mathf.Clamp((int)newValueAsVector3.y, (int)min, (int)max);
+                                newValueAsArray[2] = Mathf.Clamp((int)newValueAsVector3.z, (int)min, (int)max);
+
+                                value = newValueAsArray;
+                            }
+                            else if (value is Vector4 newValueAsVector4)
```

Comment:
Should we handle the VectorXInt as well ?

Reply:
Yeah, done